contract.php

```diff
@@ -269,3 +269,52 @@ function contract_civicrm_post($op, $objectName, $id, &$objectRef){
     }
   }
 }
+
+/**
+ * Adjust output of membership custom data for contract file download
+ * @param $content
+ * @param $context
+ * @param $tplName
+ * @param $object
+ */
+function contract_civicrm_alterContent(  &$content, $context, $tplName, &$object ) {
```

Comment:
My typical approach would be to inject some jQuery JS in the appropriate hook (e.g. ``civicrm_buildForm``), but also pass some calculated information (like whether the file exists) into the script. For a code example see https://github.com/systopia/de.systopia.moregreetings/blob/0.4.dev/moregreetings.php#L49-L55